txpool/pool.go

```diff
@@ -613,22 +613,22 @@ func (p *TxPool) AddNewGoodPeer(peerID types.PeerID) { p.recentlyConnectedPeers.
 func (p *TxPool) Started() bool                      { return p.started.Load() }
 
 func (p *TxPool) best(n uint16, txs *types.TxsRlp, tx kv.Tx, onTopOf, availableGas uint64, toSkip mapset.Set[[32]byte]) (bool, int, error) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
 	// First wait for the corresponding block to arrive
 	if p.lastSeenBlock.Load() < onTopOf {
 		return false, 0, nil // Too early
 	}
 
 	isShanghai := p.isShanghai()
+	best := p.pending.best
 
-	txs.Resize(uint(cmp.Min(int(n), len(p.pending.best.ms))))
+	txs.Resize(uint(cmp.Min(int(n), len(best.ms))))
 	var toRemove []*metaTx
 	count := 0
 
 	success, err := func() (bool, error) {
```

Comment:
I guess this closure exists only to make set lock / defer unlock scope and looks useless now

Reply:
Sure, I've removed the closure